javascript/src/URDFClasses.js

```diff
@@ -242,9 +246,81 @@ class URDFJoint extends URDFBase {
             }
 
             case 'floating':
+                // no-op if all values are identical to existing value or are null
+                if (this.jointValue.every((value, index) => values[index] === value || values[index] === null)) return didUpdate;
+                // anonymous block scope to prevent variable name clobbering with other cases in the switch
+                {
+                    // Floating joints have six degrees of freedom: X, Y, Z, R, P, Y.
+                    const [posX, posY, posZ, roll, pitch, yaw] = values;
+                    this.position.copy(this.origPosition);
+                    // Respect origin RPY when setting position
+                    if (posX !== this.jointValue[0] && posX !== null) {
+                        _tempAxis.set(1, 0, 0).applyEuler(this.rotation);
+                        this.position.addScaledVector(_tempAxis, posX);
+                        didUpdate = true;
+                    }
+                    if (posY !== this.jointValue[1] && posY !== null) {
+                        _tempAxis.set(0, 1, 0).applyEuler(this.rotation);
+                        this.position.addScaledVector(_tempAxis, posY);
+                        didUpdate = true;
+                    }
+                    if (posZ !== this.jointValue[2] && posZ !== null) {
+                        _tempAxis.set(0, 0, 1).applyEuler(this.rotation);
+                        this.position.addScaledVector(_tempAxis, posZ);
+                        didUpdate = true;
+                    }
```

Comment:
I don't think we need the if conditions here. Right now you're always resetting the object position but then you're only shifting along one of the axes if it's changed. That means the position of the joint will only reflect the values that have changed in the recent function call.

At this point in the function we can really just return `true` indicating that the joint changed since we've already checked if the joint values have changed above.

Reply:
yep, you're right - I totally blanked on the fact that the initial position set was outside the if clauses.